changes.go

```diff
@@ -0,0 +1,103 @@
+package automerge
+
+// #include "automerge.h"
+import "C"
+import (
+	"encoding/hex"
+	"runtime"
+	"time"
+)
+
+// ChangeHash is a 32-byte SHA-256 hash identifying an automerge change.
+// Like a git commit, the hash encompasses both the changes made,
+// any metadata (like commit message, or timestamp)
+// and any changes on which this change depends.
+type ChangeHash []byte
+
+// String returns the hex-encoded form of the change hash
+func (ch ChangeHash) String() string {
+	return hex.EncodeToString([]byte(ch))
+}
+
+// NewChangeHash creates a change has from its hex representation.
+func NewChangeHash(s string) (ChangeHash, error) {
+	b, err := hex.DecodeString(s)
+	if err != nil {
+		return nil, err
+	}
+	return ChangeHash(b), nil
+}
+
+// Change is a set of mutations to the document. It is analagous
+// to a commit in a version control system like Git.
+type Change struct {
+	item    *item
+	cChange *C.AMchange
+}
+
+// ActorID identifies the actor that made the change hex-encoded
+func (c *Change) ActorID() string {
+	defer runtime.KeepAlive(c)
+	return must(wrap(C.AMchangeActorId(c.cChange)).item()).actorID().String()
+}
+
+// ActorSeq is 1 for the first change by a given
+// actor, 2 for the next, and so on.
+func (c *Change) ActorSeq() uint64 {
```

Comment:
I'd prefer to keep the API minimal if possible, and avoid exposing too many types. If there was a reason to do it (e.g. if there were only a fixed number of valid values, like Kind; or we wanted a place to attach a .String() method like ChangeHash, it might make sense. But I can't think of why we'd want to do either.